src/metabase/query_processor/middleware/metrics.clj

```diff
@@ -245,14 +258,18 @@
    2. Metric source cards can reference themselves.
       A query built from a `:source-card` of `:type :metric` can reference itself."
   [query]
-  (let [query (lib.walk/walk
-               query
-               (fn [_query path-type path stage-or-join]
-                 (when (= path-type :lib.walk/join)
-                   (update stage-or-join :stages #(adjust-metric-stages query path %)))))]
-    (u/prog1
-      (update query :stages #(adjust-metric-stages query nil %))
-      (when-let [metric (lib.util.match/match-one <>
-                          [:metric _ _] &match)]
-        (log/warn "Failed to replace metric"
-                  (pr-str {:metric metric}))))))
+  (if-not (match-one-metric query)
+    query
+    (do
```

Comment:
Why not have the `try ... catch` here?

Reply:
That is what I did initially, but then I considered your comment [here](https://github.com/metabase/metabase/pull/48598#discussion_r1799965415) about being careful not to double-count errors and realized there is some very small chance that `prometheus/inc!` or (even less likely) `log/warn` could throw, in which case the error could be double-counted, but I'm probably being overly paranoid, especially since `inc!` would have to throw _after_ updating the counter to be double counted, which seems extremely unlikely. Per your comment above I don't love `adjust-metric-stages-counting-errors` either, so maybe this is a case of the cure being worse than the disease. I will update this to move the `try` / `catch` block here.